music21/stream/__init__.py

```diff
@@ -6367,36 +6376,21 @@ def updateEndMatch(n):
                 posConnected = []  # reset to empty
 
         # all results have been processed
-        # presently removing from notes, not resultObj, as index positions
-        # in result object may not be the same
         posDelete.reverse()  # start from highest and go down
 
+        for i in posDelete:
+            # environLocal.printDebug(['removing note', notes[i]])
+            # get the obj ref
+            nTarget = notes[i]
+            # Recurse rather than depend on the containers being Measures
+            # https://github.com/cuthbertLab/music21/issues/266
+            returnObj.remove(nTarget, recurse=True)
```

Comment:
This is so great.  There wasn't a `recurse=True` when we wrote this routine, and it now not only simplifies but also speeds up (probably) and removes bugs from it.  THANKS!  There are probably lots of other places that could depend on this.